### PR TITLE
[XM] Dynamically look up NSExtensionMain to unbreak pre-10.10 XM apps

### PR DESCRIPTION
**Pull request by chamons**
- So having a simple reference to NSExtensionMain is enough for /usr/lib/libextension.dylib to get added as a reference to libxammac.dylib
- If you have a reference to /usr/lib/libextension.dylib then any XM app (including our tests) bomb out on startup. This is suboptimal.
- So dlopen/dysm our way to victory.

### CODE REVIEW

File: runtime/launcher.m [hunk 1]
```diff
@@ -581,7 +581,19 @@ int xamarin_main (int argc, char **argv, bool is_extension)
...
 
 		if (is_extension) {
-			rv = NSExtensionMain (new_argc, new_argv);
+			void * libExtensionHandle = dlopen ("/usr/lib/libextension.dylib", RTLD_LAZY);
+			if (libExtensionHandle == nil)
+				exit_with_message ("Unable to load libextension.dylib", data.basename, false);
+
+			typedef int (*extension_main)(int argc, char * argv[]);
+
+			extension_main extensionMain = (extension_main)dlsym (libExtensionHandle, "NSExtensionMain");
```
Comment (rolfbjarne) [inline runtime/launcher.m]: Missing whitespace after cast: `(extension_main) dlsym`

File: runtime/launcher.m [hunk 2]
```diff
@@ -581,7 +581,19 @@ int xamarin_main (int argc, char **argv, bool is_extension)
...
+
+			if (extensionMain == nil)
+				exit_with_message ("Unable to load NSExtensionMain", data.basename, false);
+
+			rv =  (*extensionMain) (new_argc, new_argv);
```
Comment (rolfbjarne) [inline runtime/launcher.m]: Extra whitespace after equals.

**Comment by rolfbjarne** [comment]
LGTM.

**Comment by chamons** [comment]
Fixed spacing.

**Comment by xamarin-release-manager** [comment]
Build success

**Comment by xamarin-release-manager** [comment]
Build success

File: runtime/launcher.m [hunk 3]
```diff
@@ -581,7 +581,19 @@ int xamarin_main (int argc, char **argv, bool is_extension)
 		*ptr = NULL;
 
 		if (is_extension) {
-			rv = NSExtensionMain (new_argc, new_argv);
+			void * libExtensionHandle = dlopen ("/usr/lib/libextension.dylib", RTLD_LAZY);
```
Comment (spouliot) [inline runtime/launcher.m]: who's responsible for installing that dylib the `/usr/lib/` on customer's Mac ? should not it be part of the app bundle ? (e.g. versioning issues)

Comment (rolfbjarne) [inline runtime/launcher.m]: Apple ships it.

**Pull request merged**